src/openforms/contrib/kvk/validators.py

```diff
@@ -84,38 +83,43 @@ def __call__(self, value: str) -> bool:
         return True
 
 
-@register("kvk-kvkNumber", verbose_name=_("KvK number"), for_components=("textfield",))
+@register("kvk-kvkNumber")
 @deconstructible
-class KVKNumberRemoteValidator(KVKRemoteBaseValidator):
+class KVKNumberRemoteValidator(BasePlugin[str], KVKRemoteBaseValidator):
     query_param = "kvkNummer"
     value_label = _("KvK number")
 
-    def __call__(self, value, submission):
+    verbose_name = _("KvK number")
+    for_components = ("textfield",)
+
+    def __call__(self, value: str, submission):
         validate_kvk(value)
-        super().__call__(value)
+        KVKRemoteBaseValidator.__call__(self, value)
 
 
-@register("kvk-rsin", verbose_name=_("KvK RSIN"), for_components=("textfield",))
+@register("kvk-rsin")
 @deconstructible
-class KVKRSINRemoteValidator(KVKRemoteBaseValidator):
+class KVKRSINRemoteValidator(BasePlugin[str], KVKRemoteBaseValidator):
     query_param = "rsin"
     value_label = _("RSIN")
 
-    def __call__(self, value, submission):
+    verbose_name = _("KvK RSIN")
+    for_components = ("textfield",)
+
+    def __call__(self, value: str, submission):
         validate_rsin(value)
-        super().__call__(value)
+        KVKRemoteBaseValidator.__call__(self, value)
```

Comment:
Having `KVKRemoteBaseValidator` as a subclass of `BasePlugin` would require restricting `for_components` to this base class. I think using another method name instead of `__call__` might be better?

Reply:
I don't follow - the `for_components` is semantically related for plugins/components, and technically comes down to the expected type of `value`.

However, I think changing that to `KVKRemoteValidatorMixin` with a different method name than `__call__` would make sense, I do not expect this base class to be called directly/by itself.